File: backend/src/main/java/com/teamportal/service/ActivityService.java
package com.teamportal.service;

import com.teamportal.dto.ActivityDTO;
import com.teamportal.repository.ActivityRepository;
import org.springframework.stereotype.Service;

import java.time.format.DateTimeFormatter;
import java.util.List;
import java.util.stream.Collectors;

@Service
public class ActivityService {
    private final ActivityRepository activityRepository;
    private final DateTimeFormatter formatter;

    public ActivityService(ActivityRepository activityRepository) {
        this.activityRepository = activityRepository;
        this.formatter = DateTimeFormatter.ofPattern("M/d/yyyy, h:mm:ss a");
    }

    public List<ActivityDTO> getFeedActivities() {
        return activityRepository.findAllWithUser().stream()
                .map(activity -> new ActivityDTO(
                        activity.getId(),
                        activity.getAction(),
                        activity.getTimestamp().format(formatter),
                        activity.getUser() != null ? activity.getUser().getName() : null
                ))
                .collect(Collectors.toList());
    }
}

File: backend/src/main/kotlin/com/teamportal/service/ActivityService.kt
package com.teamportal.service

import com.teamportal.repository.ActivityRepository
import org.springframework.stereotype.Service
import java.time.format.DateTimeFormatter

data class ActivityDTO(
    val id: Long,
    val action: String,
    val timestamp: String,
    val userName: String? = null
)

@Service
class ActivityService(
    private val activityRepository: ActivityRepository
) {
    private val formatter = DateTimeFormatter.ofPattern("M/d/yyyy, h:mm:ss a")

    fun getFeedActivities(): List<ActivityDTO> {
        val activities = activityRepository.findAll()

        return activities.map { activity ->
            ActivityDTO(
                id = activity.id,
                action = activity.action,
                timestamp = activity.timestamp.format(formatter),
                userName = activity.user?.name
            )
        }
    }
}
